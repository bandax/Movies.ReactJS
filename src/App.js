import React from 'react';


const App = () => (
    <div>
        <h1>Welcome to Catalog Movies</h1>
    </div>
);

export default App;