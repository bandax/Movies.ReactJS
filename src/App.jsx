import React from 'react';
import { Header } from './Header';
import { Footer } from './Footer';
import { Message } from './Message';


export const App = () => (
    <div>        
        <Header />        
        <Message />
        <Footer />
    </div>
);
