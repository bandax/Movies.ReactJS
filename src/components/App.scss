* {
  box-sizing: border-box;
}

@for $i from 1 to 13 {
  col-#{$i} {
    width: 8.33 * $i#{"%"};
  }
}

[class*="col-"] {
  float: left;
}

.row::after {
  content: "";
  clear: both;
  display: table;
}

html body {
  background-color: #424242;
  padding: 20px 10px;
}

.app {
  font-size: 14px;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  color: #9ca4ab;
}

.btn {
  color: #f65261;
  padding: 5px 10px;
  border-radius: 5px;
  cursor: pointer;
}
