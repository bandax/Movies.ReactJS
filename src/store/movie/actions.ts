import * as constants from "./constants";
import { IMovie } from "../../interfaces/IResultsMovies";
import { MovieActionsTypes } from "./types";

export function loadingMovies(): MovieActionsTypes {
  return {
    type: constants.LOADING_MOVIES,
    loading: true,
  };
}

export function loadingMoviesSuccess(movies: IMovie[]): MovieActionsTypes {
  return {
    type: constants.LOAD_MOVIES_SUCCESS,
    loading: false,
    movies: movies,
  };
}

export function loadingMoviesError(errorMessage: string): MovieActionsTypes {
  return {
    type: constants.LOADING_MOVIES_ERROR,
    loading: true,
    errorMessage: errorMessage,
  };
}
export function addMovie(movie: IMovie): MovieActionsTypes {
  return {
    type: constants.ADD_MOVIE,
    movieToAdd: movie,
  };
}

export function updateMovie(movie: IMovie): MovieActionsTypes {
  return {
    type: constants.UPDATE_MOVIE,
    movieToUpdate: movie,
  };
}

export function deleteMovie(movieId: number): MovieActionsTypes {
  return {
    type: constants.DELETE_MOVIE,
    movieId: movieId,
  };
}
