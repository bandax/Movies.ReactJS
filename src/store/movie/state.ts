import { IMovie } from "../../interfaces/IResultsMovies";

export interface MovieState {
  movies: IMovie[];
  loading: boolean;
  errorMessage: string;
}
