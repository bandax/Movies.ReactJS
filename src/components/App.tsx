import * as React from "react";
import './App.scss';
import { Header } from './Header/Header';
import { Footer } from './Footer/Footer';
import { ErrorBoundary } from './Error/ErrorBoundary';
import { SubHeader } from './SubHeader/SubHeader';

import { ResultsMovie } from './ResultsMovie/ResultsMovie';
import { IMovie, IResultsMovie, IResultsMovieProps } from '../interfaces/IResultsMovies';

const movies: IMovie[] = [
  {
    id:"movie-1",
    title: "Avengers Infinity War",
    image: "../../../assets/posters/avengers-infinity-war.PNG",
    year: 2004,
    categories: [
      {
        id: "cat-1",
        name: "Action & Adventure"
      }
    ]
  },
  {
    id:"movie-2",
    title: "Pulp Fiction",
    image: "../../../assets/posters/pulp-fiction.PNG",
    year: 2004,
    categories: [
      {
        id: "cat-1",
        name: "Action & Adventure"
      },
      {
        id: "cat-2",
        name: "Drama"
      }
    ]
  }
];

const resultMovies: IResultsMovie = {
  total: 2,
  movies: movies
}

const resultsMovieProps: IResultsMovieProps = {
  title: "movies found",
  resultsMovies: resultMovies
}

export const App: React.FunctionComponent = () => (    
  <div className="app">  
    <ErrorBoundary>
      <Header />     
      <SubHeader />
      <ResultsMovie title={resultsMovieProps.title} resultsMovies={resultsMovieProps.resultsMovies} />
      <Footer />
    </ErrorBoundary>      
  </div>
);
