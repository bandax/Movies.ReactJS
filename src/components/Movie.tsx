import * as React from 'react';
import './App.scss';
import { Header } from './Header/Header';
import { ErrorBoundary } from './Error/ErrorBoundary';
import { SubHeader } from './SubHeader/SubHeader';
import ResultsMovie from './ResultsMovie/ResultsMovie';
import FooterSplit from './Footer/LoadableFooter';

// PATTERN: Composition
const Movie: React.FunctionComponent = () => (
  <div className="app">
    <ErrorBoundary>
      <Header />
      <SubHeader />
      <ResultsMovie />
      <FooterSplit />
    </ErrorBoundary>
  </div>
);

export default Movie;
