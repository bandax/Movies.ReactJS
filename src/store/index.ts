import { combineReducers, applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { movieReducer } from './movie/reducers';

const rootReducer = combineReducers({
  movie: movieReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export const store = createStore(rootReducer, applyMiddleware(thunk));
