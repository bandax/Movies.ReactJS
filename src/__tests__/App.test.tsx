import * as React from 'react';
import thunk, { ThunkDispatch } from 'redux-thunk';
import { App } from '../components/App';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import { RootState } from '../store/index';
import { MovieState } from '../store/movie/state';
import configureMockStore from 'redux-mock-store';
import { Action } from 'redux';
import { Provider } from 'react-redux';

type DispatchExts = ThunkDispatch<MovieState, void, Action>;
const middleware = [thunk];
const state: RootState = {
  movie: {
    movie: null,
    errorMessage: '',
    findMovies: true,
    loading: false,
    movies: [],
  },
};
const mockStore = configureMockStore<RootState, DispatchExts>(middleware);

describe('<App />', () => {
  let store;

  beforeEach(() => (store = mockStore(state)));

  it('should display a default app component', async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );

    const fragment = asFragment();

    expect(fragment).toMatchSnapshot();
  });
});
