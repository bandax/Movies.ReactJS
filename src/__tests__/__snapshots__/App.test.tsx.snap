// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should display a default app component 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="header row"
    >
      <div
        class="search col-10"
      >
        <span
          class="find-label"
        >
          Find your movie
        </span>
        <br />
        <input
          class="search-box"
          placeholder="What do you want to watch?"
          type="text"
          value=""
        />
        <a
          class="btn btn-search"
          href="#/search/"
        >
          Search
        </a>
      </div>
      <div
        class="add-movie col-2"
      >
        <button
          class="btn btn-add-movie"
        >
          +Add Movie
        </button>
      </div>
    </div>
    <div
      class="sub-header"
    >
      <div
        class="clasification-movies"
      >
        <ul
          class="movies-types"
        >
          <li
            class="movie-type"
          >
            <a
              href=""
            >
              Documentary
            </a>
          </li>
          <li
            class="movie-type"
          >
            <a
              href=""
            >
              Comedy
            </a>
          </li>
          <li
            class="movie-type"
          >
            <a
              href=""
            >
              Horror
            </a>
          </li>
          <li
            class="movie-type"
          >
            <a
              href=""
            >
              Action & Adventure
            </a>
          </li>
          <li
            class="movie-type"
          >
            <a
              href=""
            >
              Drama
            </a>
          </li>
          <li
            class="movie-type"
          >
            <a
              href=""
            >
              Romance
            </a>
          </li>
        </ul>
      </div>
      <div
        class="sort-movies"
      >
        <span
          class="sort-title"
        >
          sort by: 
        </span>
        <select>
          <option
            value="sort-1"
          >
            Release Date
          </option>
          <option
            value="sort-2"
          >
            Most Popular
          </option>
        </select>
      </div>
    </div>
    <div
      class="list-movies"
    >
      <div
        class="total-result"
      >
        <span
          class="found-label"
        >
          <b>
            0
          </b>
           movies found
        </span>
      </div>
      <div
        class="display-movie"
      />
    </div>
    <div
      class="footer"
    >
      <img
        src="../../../assets/netflix-roulette.PNG"
      />
    </div>
  </div>
</DocumentFragment>
`;
